### ISSUE TITLE
Improve Decentralization

### ISSUE THREAD

**Issue by ligi** [ethereum/sourcify#346]
Our main UVP is Decentralisation of the MetaData (can be seen as the main headline on our site is "Decentralized Metadata and Source Code Repository")
Yet currently we are only decentralized in a way that everyone can verify the content and hosting is provided in a decentralized manner (IPFS)
But we are still centralized in a way that we provide the dataset and currently no other entity. If we would stop the system would not work anymore.
I suggest the following - we create a PoA chain (making it on the mainnet would take up valuable state and produce significant costs which is I think one big factor we ignored this elefant in the room for a long time)
Then there is a smart-contract with rounds and a commit-reveal scheme.
In every round every participant commits to a IPFS hash (basically the IPFS hash we currently generate)
Then they reveal and get either slashed or even completely thrown out (to keep it simple) when their IPFS hash does not match the one from the majority.
In the same commit round they also provide a IPFS hash that also contains newly found contracts (either by monitoring or manual input) - this hash is used to generate the next IPFS hash (summ of all valid new contracts gathered from the last round).
The ether earned with mining on the chain is also burned when committing.
Another option would be to use another consensus PoA -> ~ PoWA where the work is the validation of the dataset.
Another option would be to use an L2 - but this would also generate costs and the question is which L2 to bind to - might lock us in.
Then we would simply invite other organisations to be validators.

**Comment by marcocastignoli** [comment]
We discussed about this problem also during DevCon and we got some new idea:
* thinking of decentralization in terms of security: the more your transaction is risky, the more you want to have a safe verification system
* we could offer three ways of fetching the metadata:
  * "centralized": the user fetches the metadata from the Sourcify's IPFS repository. (It is centralized in the way that Sourcify is the only entity managing the repository)
  * federated: the user fetches the metadata from multiple entities, if the contract is verified on all of them, the chances of the contract be really verified are very high.
  * decentralized: the user verifies the contract on the edge, it fetches the metadata using the metadata IPFS CID in the bytecode (and then the files from the metadata), and recompiles in the client. The files will be always available if the contract is verified because Sourcify is hosting them on IPFS
